nextcord/mixins.py

```diff
@@ -47,3 +57,45 @@ class Hashable(EqualityComparable):
 
     def __hash__(self) -> int:
         return self.id >> 22
+
+
+class PinsMixin:
+    __slots__ = ()
+    _state: Any
+
+    if TYPE_CHECKING:
+        _state: ConnectionState
```

Comment:
ideally just use `from __future__ import annotations` and like, not use `Any`
if youre so stuck on doing this though, you do not need a double declaration